Patch Slack Using Powershell

sheldonhull opened the pull request:
# If applied, this commit will...
Change patching behavior to use powershell instead of batch file. 

# Why is this change needed?
Prior to this change, bat file was used which was unable to do more advanced path detection, handle symlinking properly, and handle removal of the patching contents.

# How does it address the issue?
This change will 
- dynamically pick up install path.
- clear customization and repatch
- copy contents from repo directory into local user directory
- log messages using PSFramework
- remove customizations can be easily add later using the same logic if required

Tested rerunning several times and it avoided duplicating plugin injection.

sheldonhull committed:
Patch Slack Using Powershell

# If applied, this commit will...
Change patching behavior to use powershell instead of batch file. 

# Why is this change needed?
Prior to this change, bat file was used which was unable to do more advanced path detection, handle symlinking properly, and handle removal of the patching contents.

# How does it address the issue?
This change will 
- dynamically pick up install path.
- clear customization and repatch
- copy contents from repo directory into local user directory
- log messages using PSFramework
- remove customizations can be easily add later using the same logic if required

Tested rerunning several times and it avoided duplicating plugin injection.

sheldonhull committed:
Update bat file to be launcher for ps1

This will launch script as admin simplifying running

glajchs commented:
Thank you so much for this!  I haven't used Powershell at all yet and I didn't really have time to learn in the amount of time I had available to me when putting in windows support, so I was actually going to put a out a call for someone more competent in windows scripting land to do this for me, and you're right there :-)
I'll give this a proper review as soon as possible and get it on master.

sheldonhull commented:
You are welcome. I like the better security and the option to add additional plugins so figured it would find a better place here than just as a gist. Hopefully it works fine for you. If needed the exact same logic could be applied to a simple `uninstall.ps1` but honestly I don't know why anyone would uninstall :-)

sheldonhull committed:
Multiple version fixes

I had issues with latest update. 
Put a note in readme that you can run the update.exe and reset the patched files.

sheldonhull committed:
Improved handling for new versions

Improved handling for new versions in multiple paths. This correctly handled version update from app-3.3.8 to app-3.4.0.
